doc/language_reference/language_reference.md

```diff
@@ -335,6 +335,45 @@ pattern ::= (* tuple pattern *)
           | "_"             (* wildcard, matches any value *)
 ```
 
+### Automatic string conversion
+
+The compiler automatically inserts code to convert abitrary values to strings when they occur
```

Comment:
Frankly I believe that this should not appear in the language reference. Users don't need to know what the compiler does internally. Some of this discussion should be moved into a comment, but the description of the conversion function naming should be kept.

Reply:
OK, I will rephrase this to not refer to what compiler does.